configure

```diff
@@ -604,10 +600,10 @@ def configure_node(o):
   o['variables']['host_arch'] = host_arch
   o['variables']['target_arch'] = target_arch
 
-  if target_arch != host_arch and options.with_snapshot:
-    o['variables']['want_separate_host_toolset'] = 1
-  else:
+  if target_arch == host_arch and options.without_snapshot:
     o['variables']['want_separate_host_toolset'] = 0
+  else:
+    o['variables']['want_separate_host_toolset'] = 1
```

Comment:
This still doesn't look quite correct me.  Host/target toolsetting should be enabled when cross-compiling with snapshots enabled so I think you could write this as:

```python
cross_compiling = target_arch != host_arch
want_snapshots = not options.without_snapshot
o['variables']['want_separate_host_toolset'] = int(cross_compiling and want_snapshots)
```

The double negation reads kind of iffy but it is what it is.